### PR TITLE
Adds global access control model

### LINKED ISSUE #46: Add Global Access Control model and action endpoints

### PR DESCRIPTION
**Pull request by ish2028**
Closes #46

### CODE REVIEW

File: src/models/globalAccessControl.ts [hunk 1]
```diff
@@ -34,6 +35,12 @@ export interface GlobalAccessControlData {
   accessControlRights: Permission | null
   resqueRights: Permission | null
   analyticRights: AnalyticPermission | null
+  user?: GlobalAccessControlUser
+}
+
+interface GlobalAccessControlUser {
```
Comment (bayssmekanique) [inline src/models/globalAccessControl.ts:41]: Don't create new models.  Import the `user` model.

File: src/models/globalAccessControl.ts [hunk 2]
```diff
@@ -138,4 +145,11 @@ export class GlobalAccessControl extends Model {
   get analyticRights () {
     return this._globalAccessControlData.analyticRights
   }
+
+  eagerUserProfile () {
```
Comment (bayssmekanique) [inline src/models/globalAccessControl.ts:149]: This should be `eagerUser`.

**Review by bayssmekanique — CHANGES_REQUESTED** [review]

File: src/models/globalAccessControl.ts [hunk 3]
```diff
@@ -146,10 +141,10 @@ export class GlobalAccessControl extends Model {
     return this._globalAccessControlData.analyticRights
   }
 
-  eagerUserProfile () {
+  eagerUser () {
     if (typeof this._globalAccessControlData.user === 'undefined') {
       return undefined
```
Comment (bayssmekanique) [inline src/models/globalAccessControl.ts:146]: Don't use the `undefined` global.  Make this a one line ternary like the other `eager` methods.

File: src/models/user.ts [hunk 4]
```diff
@@ -127,7 +127,7 @@ export class User extends Model {
   }
 
   eagerPrimaryEmailAddress () {
-    return new PrimaryEmailAddress(this._userData.primaryEmailAddress, this.sessionHandler)
+    return typeof this._userData.primaryEmailAddress !== 'undefined' ? new PrimaryEmailAddress(this._userData.primaryEmailAddress, this.sessionHandler) : undefined
```
Comment (bayssmekanique) [inline src/models/user.ts:130]: Don't use the `undefined` global. Make this a one line ternary like the other `eager` methods.

**Review by bayssmekanique — CHANGES_REQUESTED** [review]

File: src/models/user.ts (hunk 4 above)
Comment (ish2028) [reply src/models/user.ts:130]: Did you you mean to put this issue?  This is using the one line ternary.

**Review by bayssmekanique — APPROVED** [review]

**Pull request merged**